File: java-pr/49/java/performance/ArrayProcessor28.java
// {fact rule=performance@v1.0 defects=0}
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class ArrayProcessor28 {
    private Database database = new Database();
    
    public void processArray(int[] array) {
        for (int i = 0; i < array.length; i++) {
            for (int j = i + 1; j < array.length; j++) {
                System.out.println(array[i] + array[j]);
            }
        }
    }
    
    private static class Database {
        public User findUserById(int id) { return new User(); }
        public List<User> findAllUsers() { return new ArrayList<>(); }
    }
    
    private static class User {}
}
// {/fact}

File: js-pr/56/javascript/code-quality-performance/array_processor_28.js
// {fact rule=code-quality-performance@v1.0 defects=0}
function processArray(array) {
    for (let i = 0; i < array.length; i++) {
        for (let j = i + 1; j < array.length; j++) {
            console.log(array[i] + array[j]);
        }
    }
}
// {/fact}